lib/Doctrine/DBAL/Driver/IBMDB2/DB2Statement.php

```diff
@@ -200,9 +200,11 @@ public function fetch($fetchMode = null)
         $fetchMode = $fetchMode ?: $this->_defaultFetchMode;
         switch ($fetchMode) {
             case \PDO::FETCH_BOTH:
-                return db2_fetch_both($this->_stmt);
+                $result = @db2_fetch_both($this->_stmt);
```

Comment:
There should be an error handler then, not a silence operator. This just makes it terrible to find out if something goes wrong.